RadixWallet/Features/AccountRecoveryScan/Children/AccountRecoveryScanInProgress/AccountRecoveryScanInProgress.swift

```diff
@@ -163,134 +213,153 @@ public struct AccountRecoveryScanInProgress: Sendable, FeatureReducer {
 		}
 	}
 
-	/// FIXME: This results in CancellationError, not only this but doing ANY thing that takes a bit of time inside of `scanOnLedger` results in
-	/// CancellationError, e.g. `try await Task.sleep(for: .seconds(0.5))` results in CancellationError, which results in this
-	/// Reducer never ever receiving `internal(.foundAccounts` event - aka "TCA Send" bug. I will have to write it in another manner...
-	private func performScan(accounts: IdentifiedArrayOf<Profile.Network.Account>) async throws -> (active: IdentifiedArrayOf<Profile.Network.Account>, inactive: IdentifiedArrayOf<Profile.Network.Account>) {
+	private func performScan(
+		accounts: IdentifiedArrayOf<Profile.Network.Account>
+	) async throws -> (
+		active: IdentifiedArrayOf<Profile.Network.Account>,
+		inactive: IdentifiedArrayOf<Profile.Network.Account>
+	) {
 		let accountAddresses: [AccountAddress] = accounts.map(\.address)
-		let engineAddresses: [Address] = accountAddresses.map(\.asGeneral)
 
 		do {
-			let addressOfActiveAccounts: [AccountAddress] = try await onLedgerEntitiesClient.getEntities(
-				engineAddresses,
-				[.ownerBadge, .ownerKeys],
-				nil,
-				true // force to refresh
-			).compactMap { (onLedgerEntity: OnLedgerEntity) -> AccountAddress? in
-				guard
-					let onLedgerAccount = onLedgerEntity.account,
-					case let metadata = onLedgerAccount.metadata,
-					let ownerKeys = metadata.ownerKeys,
-					let ownerBadge = metadata.ownerBadge
-				else { return nil }
-
-				func hasStateChange(_ list: OnLedgerEntity.Metadata.ValueAtStateVersion<some Any>) -> Bool {
-					list.lastUpdatedAtStateVersion > 0
-				}
-				let isActive = hasStateChange(ownerKeys) || hasStateChange(ownerBadge)
-				guard isActive else {
-					return nil
+			let activeAccounts = try await onLedgerEntitiesClient
+				.getAccounts(
+					accountAddresses,
+					metadataKeys: [.ownerBadge, .ownerKeys],
+					forceRefresh: true
+				)
+				.filter { (onLedgerAccount: OnLedgerEntity.Account) -> Bool in
+
+					guard
+						case let metadata = onLedgerAccount.metadata,
+						let ownerKeys = metadata.ownerKeys,
+						let ownerBadge = metadata.ownerBadge
+					else { return false }
+
+					func hasStateChange(_ list: OnLedgerEntity.Metadata.ValueAtStateVersion<some Any>) -> Bool {
+						list.lastUpdatedAtStateVersion > 0
+					}
+					let isActive = hasStateChange(ownerKeys) || hasStateChange(ownerBadge)
+					return isActive
 				}
-				return onLedgerAccount.address
-			}
 
 			var active: IdentifiedArrayOf<Profile.Network.Account> = []
 			var inactive: IdentifiedArrayOf<Profile.Network.Account> = []
 			for account in accounts {
-				if addressOfActiveAccounts.contains(where: { $0 == account.address }) {
+				if let activeAccount = activeAccounts.first(where: { $0.address == account.address }) {
+					var account = account
+					if let depositRule = activeAccount.details?.depositRule {
+						account.onLedgerSettings.thirdPartyDeposits.depositRule = depositRule
```

Comment:
only active accounts will have a **relevant** depositRule from GW to update with....